check_gha_pinning/__init__.py

```diff
@@ -37,12 +45,29 @@ def _build_github_url(action: str) -> str:
 
 
 def _get_action_tags(action: str) -> tuple[dict[str, str], dict[str, list[str]]]:
+    """
+    A function to get action tags along with their corresponding SHA values from a given action.
+
+    Args
+    ----
+    action : str
+             the name of the action to get tags for.
+
+    Returns
+    -------
+    by_tag : dict[str, str]
+             a mapping of tag names to their corresponding SHA values.
+    by_sha : dict[str, list[str]]
+             a mapping of SHA values to their corresponding tag names.
+    """
     repo_url = _build_github_url(action)
-    cmd = ["git", "ls-remote", "--exit-code", repo_url]
-    by_tag = {}
-    by_sha = {}
+    cmd = ["git", "ls-remote", "-t", "--exit-code", repo_url]
```

Comment:
Nit: short options are solely for devs to type quickly on the command line, scripts should always use long form opts
```suggestion
    cmd = ["git", "ls-remote", "--tags", "--exit-code", repo_url]
```